### PR TITLE
[Qt] Multisig GUI

### PR DESCRIPTION
**Pull request by rejectedpromise**
<h1>Multisignature GUI</h1>
<h4>Provides functionality which is currently only available through raw transactions. Multisignature addresses require signatures from multiple parties before coins belonging to the address are spent. Accessed through the File dropdown menu.</h4>

<h3>Address Creation</h3>
<ul>
<li>addresses allow up to 15 owners</li>
<li>from 1 to all owners may be configured as required to sign</li>
<li>with the provided redeem script, addresses can be imported by all owners after it has been created once</li>
</ul>

multisigcreate

<h3>Spending</h3>
<ul>
<li>coin control</li>
<li>multi-destination spends</li>
<li>raw input entry option</li>
<li>automatic fees</li>
<li>automatic change return</li>
</ul>

multisigspend

<h3>Signing/Committing Transaction</h3>
<ul>
<li>one click signing of transactions</li>
<li>commit button disabled until enough signatures collected</li>
<li>after enough signatures are collected, any owner of the address can commit the transaction</li>
<li>Option to sign with private keys (**Not Recommended**)</li>
</ul>

multisigsign

### CODE REVIEW

**Comment by Fuzzbawls** [comment]
Needs to have some description in the PR (with screenshots). Will also (eventually) need release notes written (these can be committed by you to the `doc/release-notes.md` file, which serves as a placeholder)

File: src/wallet.cpp [hunk 1]
```diff
@@ -1562,13 +1592,26 @@ void CWallet::AvailableCoins(vector<COutput>& vCoins, bool fOnlyConfirmed, const
...
-                    (pcoin->vout[i].nValue > 0 || fIncludeZeroValue) &&
-                    (!coinControl || !coinControl->HasSelected() || coinControl->fAllowOtherInputs || coinControl->IsSelected((*it).first, i)))
-                    vCoins.push_back(COutput(pcoin, i, nDepth,
-                        ((mine & ISMINE_SPENDABLE) != ISMINE_NO) ||
-                            (coinControl && coinControl->fAllowWatchOnly && (mine & ISMINE_WATCH_SOLVABLE) != ISMINE_NO)));
+                if (IsSpent(wtxid, i))
+                    continue;
+                if (mine == ISMINE_NO)
+                    continue;
+                if ((mine == ISMINE_WATCH_ONLY))
```
Comment (Fuzzbawls) [inline src/wallet.cpp]: Can get rid of the double parentheses here, which cause a compiler warning.

```
wallet.cpp:1599:27: warning: equality comparison with extraneous parentheses [-Wparentheses-equality]
                if ((mine == ISMINE_WATCH_ONLY))
                     ~~~~~^~~~~~~~~~~~~~~~~~~~
wallet.cpp:1599:27: note: remove extraneous parentheses around the comparison to silence this warning
                if ((mine == ISMINE_WATCH_ONLY))
                    ~     ^                   ~
wallet.cpp:1599:27: note: use '=' to turn this equality comparison into an assignment
                if ((mine == ISMINE_WATCH_ONLY))
                          ^~
                          =
```

**Comment by presstab** [comment]
ACK

**Comment by s3v3nh4cks** [comment]
utACK

**Comment by jonspock** [comment]
First screenshot the Icon color contrast is too low for the box to the left of the Address/Key. Looks blurred

**Pull request merged**